### PR TITLE
all: limit request tracer log count to five

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Currently the request tracer log is configured to retain all logs. This opens the risk of user filesystems being filled with unwanted logs. So limit the number of logs to five (picked as reasonably similar to the number of logs retained by elastic-agent, although the logs are not necessarily comparable in size or temporal domain).

Relevant docs for the change:
- https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#_request_tracer_maxbackups
- https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#_resource_tracer_maxbackups

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

**Comment by ebeahan** [comment]
So the configuration would now limit to 5 request tracer logs, each 1MB in size per the (`resource.tracer.maxsize`)[https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#_resource_tracer_maxsize] default?

**Comment by efd6** [comment]
Yes, that's approximately correct. There are a couple of integrations that limit log size to 5MB in their configs here, but all are now limited to 5 files.

**Review by ebeahan — APPROVED** [review]
++

Initially, I worried slightly about limiting to 5MB being too limited for certain troubleshooting situations. However, I think it's a sensible starting point and better than inadvertently filling disk if the toggle is turned on and forgotten. 😅

File: packages/hashicorp_vault/changelog.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@
   changes:
     - description: Update package spec to 2.9.0.
       type: enhancement
-      link: https://github.com/elastic/integrations/pull/1
+      link: https://github.com/elastic/integrations/pull/8489
```
Comment (leehinman) [inline packages/hashicorp_vault/changelog.yml:59]: Is this right PR number for this change?

File: packages/o365/changelog.yml [hunk 2]
```diff
@@ -36,7 +41,7 @@
   changes:
     - description: Fix mappings for dynamically mapped fields.
       type: enhancement
-      link: https://github.com/elastic/integrations/pull/1
+      link: https://github.com/elastic/integrations/pull/8489
```
Comment (leehinman) [inline packages/o365/changelog.yml:44]: Is this the right PR number for this change?

**Review by leehinman — APPROVED** [review]
Changes to httpjson.yml.hbs look good, but I did notice 2 changelog.yml changes that looked odd.

File: packages/hashicorp_vault/changelog.yml (hunk 1 above)
Comment (efd6) [reply packages/hashicorp_vault/changelog.yml:59]: No, no it is not. I will repair that to what it should have been.

File: packages/o365/changelog.yml (hunk 2 above)
Comment (efd6) [reply packages/o365/changelog.yml:44]: Likewise.

**Review by muthu-mps — COMMENTED** [review]
If we are setting a default limit for the request tracer `maxbackups`. Can we update the documentation to have a mention about the default limit [here](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#_request_tracer_maxbackups:~:text=are%20retained%20indefinitely.-,request.tracer.maxbackups,-edit)? 
Otherwise change looks good!

**Review by muthu-mps — APPROVED** [review]

**Comment by efd6** [comment]
@muthu-mps The default limit in filebeat is unchanged. This change only affects the integrations.

**Review by zmoog — APPROVED** [review]

**Pull request merged**